client/telemetry/src/lib.rs

```diff
@@ -123,6 +123,14 @@ impl TelemetryEndpoints {
 	}
 }
 
+impl std::ops::Deref for TelemetryEndpoints {
+	type Target = Vec<(Multiaddr, u8)>;
+
+	fn deref(&self) -> &Self::Target {
+		&self.0
+	}
+}
+
```

Comment:
Maybe this instead, which does not expose implementation details.

```suggestion
impl TelemetryEndpoints {
    pub fn is_empty(&self) -> bool {
        self.0.is_empty()
    }
}

```

Reply:
Fine by me! Tbh I'm not entirely sure this struct is really useful nor if it should be here at all. The only thing it does is help parsing String and convert it to (multiaddr, u8)... this sounds strangely like it should be in sc-cli and not here at all.